core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -48,6 +48,12 @@
 #include <cstdint>
 #include <string>
 
+#if defined(KOKKOS_ENABLE_OPENMPTARGET) && defined(KOKKOS_ENABLE_CXX17)
+#define KOKKOS_IMPL_IF_ON_HOST if constexpr (omp_is_initial_device() == true)
+#else
+#define KOKKOS_IMPL_IF_ON_HOST if (true)
+#endif
```

Comment:
#undef it at the end of the file and comment you do so

Reply:
done